### PR TITLE
Request/Response logging

### PR DESCRIPTION
**Pull request by tiopramayudi**
**What this PR does / why we need it**:
This PR aims to enable users deploy version endpoint with request/response logging feature. This PR only touch API and SDK implementation. UI implementation will be raised in separate PR

**Does this PR introduce a user-facing change?**:
None

**Checklist**

- [x] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [x] Update Swagger spec if the PR introduce API changes
- [x] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

**Review by pradithya — APPROVED** [review]
LGTM

File: api/models/logger.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	Model          *LoggerConfig `json:"model"`
+	Transformer    *LoggerConfig `json:"transformer"`
+}
+
+type LoggerConfig struct {
+	Enabled bool       `json:"enabled"`
+	Mode    LoggerMode `json:"mode"`
+}
+
+func GetLoggerMode(mode LoggerMode) kfsv1alpha2.LoggerMode {
```
Comment (ariefrahmansyah) [inline api/models/logger.go:44]: nitpick: using GetLoggerMode is confusing for me when the object returned is KFServing's LoggerMode. i prefer to rename it to `ToKFServingLoggerMode()`

File: api/models/logger.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	modeMapping := map[LoggerMode]kfsv1alpha2.LoggerMode{
```
Comment (ariefrahmansyah) [inline api/models/logger.go:45]: nitpick: we can move this map object outside of the function so it will only initialize once

**Review by ariefrahmansyah — APPROVED** [review]
I have some nitpick comments. But overall, LGTM

**Pull request merged**